client/x/evmengine/keeper/keeper.go

```diff
@@ -172,6 +172,11 @@ func (k *Keeper) parseAndVerifyProposedPayload(ctx context.Context, msg *types.M
 //
 // Note that the validator set can change, so this is an optimistic check.
 func (k *Keeper) isNextProposer(ctx context.Context, currentProposer []byte, currentHeight int64) (bool, error) {
+	// cometAPI is lazily set and may be nil on startup (e.g. replayBlocks).
```

Comment:
I think cometAPI is set during node start. It doesn't seem to be lazily set.

Reply:
By ‘lazily set,’ I meant that, as you can see in the following code, cometAPI is only set after `newCometNode` finishes, which includes the process of replaying blocks (if any). 
During this replay, the postFinalizeBlock is called, but at that point, cometAPI is still nil.

https://github.com/piplabs/story/blob/9603826b463258e4a2033eff271ba8aae7ca33c0/client/app/start.go#L133-L140

I will update the comments to clarify this.

Reply:
@edisonz0718 
updated comment in https://github.com/piplabs/story/pull/128/commits/6cb1e6da039e6ab998ca311b22d9488e0f2babbb

Reply:
Ok make sense